### PR TITLE
resurface import errors into cluster operator condidtions (phase one,…

### PR DESCRIPTION
**Pull request by gabemontero**
… prior to introducing new condition)

Some tweaks that resurface import errors into cluster operator conditions based on the meeting between @bparees @smarterclayton and myself

These changes are a prelim to the acceptance of the new condition requested by @smarterclayton via https://github.com/openshift/api/pull/355

### CODE REVIEW

**Comment by gabemontero** [comment]
/assign @bparees 

@openshift/openshift-team-developer-experience fyi

File: pkg/apis/samples/v1/types.go [hunk 1]
```diff
@@ -328,10 +328,12 @@ func (s *Config) ClearNameInReason(reason, name string) string {
...
-	noInstallDetailed = "Samples installation in error at %s: %s"
-	installed         = "Samples installation successful at %s"
-	moving            = "Samples processing to %s"
-	removing          = "Deleting samples at %s"
+	noInstallDetailed  = "Samples installation in error at %s: %s"
+	installed          = "Samples installation successful at %s"
+	moving             = "Samples processing to %s"
+	removing           = "Deleting samples at %s"
+	doneImportsFailed  = "Samples installed at %s, with image import failures, see samples operator config object"
+	failedImageImports = "FailedImageImports"
```
Comment (bparees) [inline pkg/apis/samples/v1/types.go]: "Progressing=false, reason=FailedImageImports" still strikes me as a slightly weird thing to report to a user, but if @smarterclayton says that's the preferred to to bubble up the information, i guess this accomplishes that.

Comment (gabemontero) [reply pkg/apis/samples/v1/types.go]: yeah @bparees my recollection from out discussion was that the reason field on progressing is what was highlighted in the console

Though I have not been able to find that exactly, so maybe I'm off here

Comment (gabemontero) [reply pkg/apis/samples/v1/types.go]: bump @bparees @smarterclayton 

please confirm if this form of reporting was in fact one of the to-do's for failed imagestream import notification that came out of our disconnected discussions a couple of weeks ago.

Comment (bparees) [reply pkg/apis/samples/v1/types.go]: i think the immediate to do that came out of the meeting was to report this via a new condition on the clusteroperator resource (ie the AttentionRequired condition), but I see no reason not to also have this condition on the samples operator resource, and then use it to drive the update of the clusteroperator resource once we agree on the AttentionRequired condition.

tldr this seems fine.
